### ISSUE TITLE
ReaderListeners are not picked up

### ISSUE THREAD

**Issue by danigulyas** [swagger-api/swagger-core#2270]
Hi!

Good job with the package, it's real cool 🕶 !:)

Small bug i've found while trying to implement `ReaderListener`s, please see details below:

**Prerequisites:**
Have a class in the scanned resources package(s) which implements `ReaderListener` without any  annotation on it.

**Expected behavior:**
The class is picked up, instantiated, then acted upon accordingly to the [documentation](https://github.com/swagger-api/swagger-core/wiki/Annotations-1.5.X#customising-the-swagger-definition).

**Actual behavior:**
The class is not picked up, nor called.

**Workaround:**
The class is picked up when it's annotated with`@SwaggerDefinition`.

Thanks for making this project, enjoy your day!:)
Reg,
-Dan

**Comment by webron** [comment]
Did you add the `ReaderListener` to your scanned files? I recall someone with a similar issue not too long ago, where this was the problem.

Can you share your current configuration?

**Comment by danigulyas** [comment]
Hi Webron!

I'm using `dropwizard-swagger`, the package the class was in were in the "resource packages".

**Comment by webron** [comment]
Do you mean https://github.com/federecio/dropwizard-swagger?

**Comment by danigulyas** [comment]
Yes, a [fork](https://github.com/smoketurner/dropwizard-swagger) of that.

**Comment by seanadkinson** [comment]
I had the same issue with dropwizard and swagger v3.  Adding `@OpenAPIDefinition` to my class implementing `ReaderListener` did the trick.

**Comment by salilkumarjain** [comment]
Hi, I am also facing similar issue while working with a sample project downloaded from
https://www.javainuse.com/spring/boot_swagger
with just one API definition.

Have updated the following dependencies in pom.xml
spring-boot-starter-parent:v2.3.4.RELEASE, springfox-swagger2:v2.9.2, springfox-swagger-ui:v2.9.2, swagger-jaxrs:v1.6.2

I have added on class which implements ReaderListener and annotated with @SwaggerDefinition
Have implemented the beforeScan and afterScan methods by just putting a sysout.
On running the application the sysouts never get printed

I tried with both @SwaggerDefinition and @OpenAPIDefinition but could not get it t work. 
Have also tried the following but it seems to be not working
https://stackoverflow.com/questions/39856511/swagger-readerlistener-is-never-picked-up-during-scanning
https://github.com/swagger-api/swagger-core/issues/1912

Please help.

**Comment by MichakrawSB** [comment]
Closing this issue as part of our https://github.com/swagger-api/swagger-core/issues/5074.
This issue is outdated and/or inactive for a long time, and we’re not able to revisit it meaningfully.

If this problem is still relevant, please open a new issue and reference this one for context.
Thank you for your understanding!
